docs/src/modules/java/pages/serialization.adoc

```diff
@@ -0,0 +1,157 @@
+= Serialization
+:page-aliases:spring:serialization.adoc, java:serialization.adoc,
+
+== Jackson serialization
+
+In the Kalix Java SDK, you do not need to create serializers for the messages, events, or the state of Kalix components. The same is true for Kalix endpoints. Kalix exposes the inputs and outputs of endpoints as JSON, but you need to make them serializable with https://github.com/FasterXML/jackson[Jackson]. There are two ways to do this.
+
+1. If you are using Java https://openjdk.org/jeps/395[record] then no annotation is needed. It just works. It's as simple as using `record` instead of `class`. Kalix leverages https://github.com/FasterXML/[Jackson] under the hood and makes these records serializable for you.
+2. If you are using Java `class` then you need to annotate them with the https://github.com/FasterXML/jackson-annotations#usage-general[proper Jackson annotation].
+
+Kalix uses a predefined `Jackson` configuration, for serialization. Use the `JsonSupport` utility to update the `ObjectMapper` with your custom requirements.
```

Comment:
A mention of the `-parameters` compiler flag somewhere would be good (even if we probably include that in all samples/quickstarts already)

Reply:
added